Mobile: Add native Android trit methods for address gen and signatures (includes #1071)

cvarley100 opened the pull request:
# Description

- Add updated Entangled so files with signature/address gen trit methods
- Update Android native modules to utilise trit-based methods
- Update js address gen function to include necessary trit conversions

Includes #1071 

## Type of change

- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Tested on Android sim

# Checklist:

_Please delete items that are not relevant._

- [ ] My code follows the style guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes

cvarley100 committed:
Mobile: Add new Entangled trit methods and intergrate Android

cvarley100 committed:
Mobile: Use correct bundle hash encoding for native signatures in Android

rajivshah3 reviewed (changes requested):
`libjsc.so` and `libc++_shared.so` probably shouldn’t be checked into git. I’ll make a PR to gitignore these

laumair commented on src/mobile/android/app/src/main/java/org/iota/mobile/Converter.java:
```diff
@@ -0,0 +1,27 @@
+import com.facebook.react.bridge.WritableNativeArray;
+
+import java.util.ArrayList;
+import java.util.List;
```
Unused imports.

laumair commented on src/mobile/android/app/src/main/java/org/iota/mobile/EntangledAndroid.java:
```diff
@@ -4,12 +4,21 @@
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
```
Unused imports.

Commit:
Merge branch 'feature/realm' into feat/trits-entangled

Commit:
Batched proof-of-work (#1071)

* Rebuild entangled android

Commit used: https://github.com/iotaledger/entangled/pull/810/commits/84f744670b2b96bf8e8f1dbc95f2d89e25e6140c
PR: https://github.com/iotaledger/entangled/pull/810

* Add batched proof-of-work methods in EntangledAndroid native module

* Integrate entangled batched proof-of-work methods

* Fix failing tests

Commit:
Fix invalid address data issue (#1089)

Interrupting new account onboarding (on loading screen) leads wallet to throw continuous exceptions. The reason for exceptions was missing "completed" property in realm schema. This commit fixes the issue and also adds realm migration from schema version 0 to latest schema.

Commit:
Fix invalid bundle issue on zero value transaction with bundle size > 1 (#1093)
